FileIO/MPI_IO/NodePartitionedMeshReader.h

```diff
@@ -0,0 +1,252 @@
+/*!
+  \file NodePartitionedMeshReader.h
+  \author Wenqing Wang
+  \date   2014.08
+  \brief  Declare a class to read node-wise partitioned mesh with MPI functions.
+
+  \copyright
+  Copyright (c) 2012-2014, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+ */
+
+#ifndef NODE_PARTITIONED_MESH_READER_H
+#define NODE_PARTITIONED_MESH_READER_H
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include <mpi.h>
+
+#include "MeshLib/NodePartitionedMesh.h"
+
+namespace MeshLib
+{
+class Node;
+class Element;
+}
+
+namespace FileIO
+{
+/// Class for parallel reading of ascii or binary partitioned mesh files into a
+/// NodePartitionedMesh via MPI.
+class NodePartitionedMeshReader
+{
+public:
+    ///  \param comm   MPI communicator.
+    NodePartitionedMeshReader(MPI_Comm comm);
+
+    ~NodePartitionedMeshReader();
+
+    /*!
+         \brief Create a NodePartitionedMesh object, read data to it,
+                and return a pointer to it. Data files are either in
+                ASCII format or binary format.
+         \param file_name_base  Name of file to be read, and it must be base name without name extension.
+         \return                Pointer to Mesh object. If the creation of mesh object
+                                fails, return a null pointer.
+     */
+    MeshLib::NodePartitionedMesh* read(const std::string &file_name_base);
+
+private:
+    /// Pointer to MPI commumicator;
+    MPI_Comm _mpi_comm = MPI_COMM_WORLD;
```

Comment:
is it necessary to set initial value here? the variable is anyway overwritten in the constructor.